src/api.rs

```diff
@@ -205,12 +205,16 @@ impl BackendApi for GoApi {
             gas_limit,
             print_debug,
         };
-        let callee_instance = match cache.get_instance(&checksum, backend, options) {
+        let mut callee_instance = match cache.get_instance(&checksum, backend, options) {
             Ok(ins) => ins,
             Err(e) => return (Err(BackendError::unknown(e.to_string())), gas_info),
         };
-
         callee_instance.env.set_serialized_env(&contract_env);
+        callee_instance.set_storage_readonly(caller_env.is_storage_readonly());
+        match caller_env.pass_callstack(&mut callee_instance.env) {
```

Comment:
it will be changed to calling the `try_pass_callstack `after merged cosmwasm PR(https://github.com/line/cosmwasm/pull/178).